scripts/zkai_native_block_boundary_pivot_selector_gate.py

```
@@ -0,0 +1,696 @@
+    return json.dumps(data, sort_keys=True, separators=(",", ":"), ensure_ascii=True, allow_nan=False)
+
+
+def commitment(data: Any) -> str:
+    return "blake2b-256:" + hashlib.blake2b(canonical_json(data).encode("utf-8"), digest_size=32).hexdigest()
+
+
+def load_json(path: pathlib.Path) -> dict[str, Any]:
+    with path.open("r", encoding="utf-8") as handle:
+        payload = json.load(handle)
+    if not isinstance(payload, dict):
+        raise PivotSelectorError(f"{path} must contain a JSON object")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)